src/VisualStudio/Core/Def/LanguageService/AbstractPackage`2.cs

```diff
@@ -64,11 +55,28 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             return _languageService.ComAggregate;
         });
 
-        await shell.LoadPackageAsync(Guids.RoslynPackageId);
+        var shell = await GetServiceAsync<SVsShell, IVsShell7>(throwOnFailure: true, cancellationToken).ConfigureAwait(true);
+
+        await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+        foreach (var editorFactory in CreateEditorFactories())
+        {
+            RegisterEditorFactory(editorFactory);
+        }
 
         var miscellaneousFilesWorkspace = this.ComponentModel.GetService<MiscellaneousFilesWorkspace>();
         RegisterMiscellaneousFilesWorkspaceInformation(miscellaneousFilesWorkspace);
 
+        await shell.LoadPackageAsync(Guids.RoslynPackageId);
```

Comment:
I know we're already doing this, but would want to know why we need to do this.

Reply:
I think it's necessary, but it's maybe something which could be improved. Is there a way to specify inter-package dependencies other than manually loading like this?

Reply:
Package loads generally shouldn't have side-effects, so any behavior its depending on should be expressed in a form that is more explicit such as a service retrieval.